Automattic/jetpack
# Paid blocks: Show upgrade nudge for paid blocks nested inside a cover block

Fixes https://github.com/Automattic/jetpack/issues/26231

#### Changes proposed in this Pull Request:

Fixes a regression introduced by https://github.com/Automattic/jetpack/pull/25812 that prevented the upgrade nudges of paid blocks nested inside a cover block from being visible.

Before | After
--- | ---
<img width="666" alt="Screen Shot 2022-09-15 at 14 23 28" src="https://user-images.githubusercontent.com/1233880/190403427-55ac86ca-9a85-4952-b4fe-2c70629a754e.png"> | <img width="637" alt="Screen Shot 2022-09-15 at 14 21 46" src="https://user-images.githubusercontent.com/1233880/190403449-e1093416-0f4f-412d-8795-596e28d77e04.png">

It also improves the placement of the upgrade nudge displayed for the cover block after trying to upload a video.

Before | After
--- | ---
<img width="679" alt="Screen Shot 2022-09-15 at 14 18 22" src="https://user-images.githubusercontent.com/1233880/190403660-e409f437-e992-48e7-b989-baeb78ac9ee7.png"> | <img width="671" alt="Screen Shot 2022-09-15 at 14 21 19" src="https://user-images.githubusercontent.com/1233880/190403693-e40c6456-ad21-43c1-992f-d6d7dcdf1fcb.png">



#### Other information:


#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Apply the diff created Fusion to your WP.com sandbox
- Sandbox a WP.com simple site with a Free plan
- Create a post
- Insert a Cover block
- Try uploading a video
- Make sure the upgrade nudge is placed above the block
- Either upload an image or select a background color
- Insert a paid block such as the Payment buttons block inside the cover block
- Make sure the upgrade nudge for the nested block shows up
